client/bridge_vlan.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"github.com/ddelnano/terraform-provider-mikrotik/client/internal/types"
+	"github.com/go-routeros/routeros"
+)
+
+// BridgeVlan defines vlan filtering in bridge resource
+type BridgeVlan struct {
+	Id       string                `mikrotik:".id"`
+	Bridge   string                `mikrotik:"bridge"`
+	Tagged   types.MikrotikList    `mikrotik:"tagged"`
+	Untagged types.MikrotikList    `mikrotik:"untagged"`
+	VlanIds  types.MikrotikIntList `mikrotik:"vlan-ids"`
+}
+
+var _ Resource = (*BridgeVlan)(nil)
```

Comment:
What does this accomplish? I cloned your branch locally and everything compiles and runs correctly without it.

Reply:
It is a way to force interface implementation, a reminder in case `Resource` interface changes but you forget to update the struct. If that happens, you will get an error on this line that `cannot use *BridgeVlan as Resource: missing method ...` (you can check it by simple commenting `func (b *BridgeVlan) IDField() string {}` function)
Also, it is useful if you have optional interface like [`Finder`](https://github.com/ddelnano/terraform-provider-mikrotik/blob/374cae8383e8fd53ad50ce58ac226147a1bee561/client/client_crud.go#L47) which also could be tested with `var _ Finder = (*BridgeVlan)(nil)`


Anyway, it is a defensive tactic, we can safely remove it. If we have problems in the future, we can bring it back.
Please, feel free to remove

Reply:
ah, I see now. Using it sounds good to me, I just wanted to make sure I understood its purpose.